spark-3.4/clickhouse-spark/src/main/scala/xenon/clickhouse/func/Util.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package xenon.clickhouse.func
+
+object Util {
+  def intHash64Impl(x: Long): Long =
```

Comment:
do we need to consider the difference between big-endian and little-endian?

Reply:
For string handling in this PR, as you mentioned because UTF8 has a consistent layout, the code is fine (and tested). For other types, this could be an issue and I did not investigate very much. This is one of the complexities for supporting other types in the future, and why I am not supporting it here.

Reply:
here my concern is the memory layout of `x`

Reply:
I think it would be ok as long as we handle either encoding byte arrays to long correctly, in this case, UTF8 -> byte array -> hash (long type).

Reply:
Endianness should affect byte-level manipulation, and operations like bit shift on integer types should not be affected, so `x`'s memory layout should be irrelevant inside `intHash64Impl` and other similar function's implementation (although Clickhouse uses uint and java only has signed int, so this needs to be taken carefully). We need to make sure that the hash function behaves correctly when it involves with byte array. In this PR's scope, we only handle UTF8 string, and its pipeline is tested to be correct.

Reply:
it sounds reasonable, thanks for your in-depth analysis.